hal/src/nRF52840/core_hal.c

```diff
@@ -938,6 +945,9 @@ int HAL_Core_Execute_Standby_Mode_Ext(uint32_t flags, void* reserved) {
     if (flags & HAL_STANDBY_MODE_FLAG_DISABLE_WKP_PIN) {
         return SYSTEM_ERROR_NOT_SUPPORTED;
     }
+ 
+    // Disable thread scheduling
+    os_thread_scheduling(false, NULL);
```

Comment:
This shouldn't be done. You are disabling scheduling and later on attempt to acquire the exflash lock which is implemented using a recursive mutex. If by this point it's being held by some other thread, it will never be released and everything will deadlock. `sd_nvic_critical_region_enter` will also disable thread scheduling, disabling the scheduler is unnecessary.